### PR TITLE
Websocket subagents 0.4.3

### PR DESCRIPTION
**Pull request by athola**

### CODE REVIEW

**Comment by athola** [comment]
## Test Plan for PR #70

Generated from `/pr-review` on 2026-01-01

### Prerequisites
- [x] All blocking issues addressed (none identified)
- [x] All in-scope issues addressed (all requirements met)
- [x] Code compiles without errors

---

### Feature Verification

#### HTTP Transport Feature
**Files:** `crates/server/src/http_transport.rs`, `crates/server/src/commands/serve.rs`
**Requirement:** Remote MCP access via HTTP

**Verification Steps:**
1. [x] Review HTTP transport implementation at `http_transport.rs`
2. [x] Verify feature-gating with `#[cfg(feature = "http-transport")]`
3. [ ] Run: `cargo test -p skrills-server --features http-transport -v`
4. [ ] Run integration test: `cargo test -p skrills-server --test http_transport_integration --features http-transport`
5. [ ] Manual check: Start server with `--http 127.0.0.1:8080` and verify binding

**Expected Outcome:**
- Feature only compiles when `http-transport` feature enabled
- Integration test passes (server binds to address)
- HTTP endpoint responds to MCP requests

---

#### Settings Module Extraction
**File:** `crates/subagents/src/settings.rs`
**Requirement:** Clean SRP extraction of configuration handling

**Verification Steps:**
1. [x] Review settings module structure
2. [x] Verify ExecutionMode parsing tests exist
3. [ ] Run: `cargo test -p skrills-subagents settings -v`
4. [ ] Verify config file paths: `~/.claude/subagents.toml` and `~/.codex/subagents.toml`

**Expected Outcome:**
- Unit tests pass for execution mode parsing
- Backend parsing tests pass
- Config loading handles missing files gracefully

---

#### Tool Schemas Module Extraction
**File:** `crates/subagents/src/tool_schemas.rs`
**Requirement:** Clean SRP extraction of MCP tool definitions

**Verification Steps:**
1. [x] Review tool_schemas module structure
2. [ ] Run: `cargo test -p skrills-subagents tool_schemas -v`
3. [ ] Verify `all_tools()` returns expected tool definitions

**Expected Outcome:**
- Tool schemas correctly define run_subagent, cancel_subagent, and list_runs
- All required fields populated

---

### Build & Quality Gates

**Run these commands to verify overall quality:**

```bash
# Full test suite
cargo test --workspace

# With HTTP transport feature
cargo test --workspace --features http-transport

# Linting
cargo clippy --workspace --all-targets --features http-transport

# Build verification
cargo build --workspace --features http-transport
```

**All must pass before PR approval.**

---

### Summary Checklist

| Component | File | Verified | Notes |
|-----------|------|----------|-------|
| HTTP Transport | http_transport.rs | [x] | Feature-gated, integration test exists |
| Serve Command | serve.rs | [x] | HTTP mode properly integrated |
| Settings Module | settings.rs | [x] | Unit tests included |
| Tool Schemas | tool_schemas.rs | [x] | Extracted cleanly |
| Store | store.rs | [x] | Minor formatting changes only |
| Service | service.rs | [x] | Uses extracted modules |

**Ready for merge when all boxes checked.**

---
*Test plan generated by /pr-review - execute with /fix-pr*

**Comment by athola** [comment]
## PR Review Feedback Addressed

All verification steps from the test plan have been executed.

### Test Plan Execution Results

| Component | Verification | Status |
|-----------|--------------|--------|
| HTTP Transport | `cargo test -p skrills-server --features http-transport` | ✅ PASS (159 tests + 1 integration) |
| Settings Module | `cargo test -p skrills-subagents settings` | ✅ PASS |
| Tool Schemas | `cargo test -p skrills-subagents tool_schemas` | ✅ PASS |
| Full Workspace | `cargo test --workspace --features http-transport` | ✅ PASS |

### Quality Gates

| Check | Status |
|-------|--------|
| `cargo test --workspace --features http-transport` | ✅ All tests pass |
| `cargo clippy --workspace --all-targets --features http-transport` | ✅ No warnings |
| `cargo build --workspace --features http-transport` | ✅ Build successful |

### Coverage Analysis

Codecov reports 92.68% patch coverage with 36 lines missing coverage. The uncovered areas are:
- `settings.rs` (75%) - Config file loading paths (acceptable - filesystem operations)
- `serve.rs` (0% for 2 lines) - HTTP mode server startup code (difficult to unit test)
- `scorer.rs`, `tokens.rs`, `analytics.rs` - Edge cases in analytics code

This coverage level is acceptable for a feature-flagged transport layer.

### Issue Linkage

| Issue | Title | Status | Action |
|-------|-------|--------|--------|
| #25 | WebSocket/HTTP control-plane streaming | Partially Addressed | Commented (Phase 1 complete, WebSocket deferred) |
| #69 | Auth and TLS for HTTP transport | Not affected | Follow-up issue for Phase 2 security |

### Review Thread Status

No unresolved review threads - PR is ready for merge.

---

Ready for final approval. All pre-commit hooks pass.

File: crates/server/src/cli.rs [hunk 1]
```diff
@@ -95,6 +95,11 @@ pub enum Commands {
         /// Watches filesystem for changes and invalidates caches immediately.
         #[arg(long, default_value_t = false)]
         watch: bool,
+        #[cfg(feature = "http-transport")]
```
Comment (athola) [inline crates/server/src/cli.rs]: this should be configurable at runtime, not at compile time

**Comment by athola** [comment]
## Test Plan v2 Execution Results

Executed: 2026-01-02 | Commits verified: `bae8732`, `9c973d1`, `690f03b`, `214e409`

---

### 1. HTTP Transport Installation (`--http` flag)

| Verification | Status | Notes |
|--------------|--------|-------|
| Installer `--http` argument parsing (`install.sh:297-304`) | ✅ PASS | Correctly parses `--http <addr>` and `--http=<addr>` forms |
| `install_systemd_service()` function (`install-skrills.sh:366-432`) | ✅ PASS | Creates valid systemd unit file with proper restart behavior |
| MCP registration with HTTP URL | ✅ PASS | Supports both `claude mcp add` and manual `.mcp.json` fallback |
| Script syntax validation | ✅ PASS | `bash -n` passes on both scripts |

---

### 2. Runtime HTTP Transport Configuration

| Verification | Status | Notes |
|--------------|--------|-------|
| `--http` appears in CLI help | ✅ PASS | Always visible regardless of compile-time feature |
| Help text clarifies feature requirement | ✅ PASS | States "Requires the \`http-transport\` feature (enabled by default)" |
| Runtime error when feature not compiled | ✅ PASS | Code path exists at `serve.rs:45-51` |
| HTTP server binds successfully | ✅ PASS | `timeout 2 skrills serve --http 127.0.0.1:9876` runs without error |

---

### 3. HTTP Transport Default in Release Binaries

| Verification | Status | Notes |
|--------------|--------|-------|
| CLI `Cargo.toml` default features | ✅ PASS | `default = ["subagents", "http-transport"]` |
| Server `Cargo.toml` default features | ✅ PASS | `default = ["watch", "subagents", "http-transport"]` |
| Release build | ✅ PASS | `cargo build --release` succeeds |
| Release binary supports `--http` | ✅ PASS | Server starts on specified address |

---

### 4. Documentation Updates

| Component | Status | Notes |
|-----------|--------|-------|
| README.md HTTP section | ✅ PASS | "Installation with HTTP Transport" section present |
| CHANGELOG.md 0.4.3 entries | ✅ PASS | Documents all three HTTP-related changes |
| Makefile `demo-http` target | ✅ PASS | Target exists with proper help text |

---

### 5. Regression Testing

```bash
cargo test --workspace                                    # ✅ ALL PASS
cargo test --workspace --features http-transport          # ✅ ALL PASS  
cargo clippy --workspace --all-targets --features http-transport -- -D warnings  # ✅ NO WARNINGS
cargo test -p skrills-server --test http_transport_integration --features http-transport  # ✅ 1/1 PASS
```

---

### Summary

| Component | Status |
|-----------|--------|
| `--http` installer flag | ✅ Verified |
| Systemd service install | ✅ Verified |
| Runtime HTTP config | ✅ Verified |
| HTTP default in release | ✅ Verified |
| Documentation | ✅ Verified |
| All tests pass | ✅ Verified |

**Ready for merge.** All Test Plan v2 verification steps complete.
